conf/docker/dataverse-glassfish/init-container/populate

```diff
@@ -0,0 +1,307 @@
+use Getopt::Long;
+use Socket;
+use File::Copy;
+#for timing purposes
+
+my $verbose;
+my $pg_only;
+my $force;
+
+my @CONFIG_VARIABLES;
+
+my $postgresonly = 0;
+
+    @CONFIG_VARIABLES =
+      ( 'POSTGRES_SERVER', 'POSTGRES_PORT', 'POSTGRES_DATABASE', 'POSTGRES_USER', 'POSTGRES_PASSWORD', 'POSTGRES_ADMIN_PASSWORD' );
+    
+
+my %CONFIG_DEFAULTS = (
+
+    'POSTGRES_ADMIN_PASSWORD', 'secret',
+    'POSTGRES_SERVER',   '127.0.0.1',
+    'POSTGRES_PORT',     5432,
+    'POSTGRES_DATABASE', 'dvndb',
+    'POSTGRES_USER',     'dvnapp',
+    'POSTGRES_PASSWORD', 'secret',
+);
+
+$CONFIG_DEFAULTS{'POSTGRES_SERVER'} = $ENV{"POSTGRES_SERVER"} . "." .  $ENV{"POSTGRES_SERVICE_HOST"};
+$CONFIG_DEFAULTS{'POSTGRES_DATABASE'} = $ENV{"POSTGRES_DATABASE"};
+$CONFIG_DEFAULTS{'POSTGRES_USER'} = $ENV{"POSTGRES_USER"};
+
+my %CONFIG_PROMPTS = (
+    'POSTGRES_SERVER',   'Postgres Server Address',
+    'POSTGRES_PORT',     'Postgres Server Port',
+    'POSTGRES_ADMIN_PASSWORD', 'Postgres ADMIN password',
+    'POSTGRES_DATABASE', 'Name of the Postgres Database',
+    'POSTGRES_USER',     'Name of the Postgres User',
+    'POSTGRES_PASSWORD', 'Postgres user password',
+);
+
+# Supported Posstgres JDBC drivers:
+# (have to be configured explicitely, so that Perl "taint" (security) mode
+# doesn't get paranoid)
+# 3. CHECK POSTGRES AND JQ AVAILABILITY:
+
+my $pg_local_connection = 0;
+my $psql_exec;
+my $jq_exec = "";
+
+my $POSTGRES_SYS_UID;
+if ( $CONFIG_DEFAULTS{'POSTGRES_SERVER'} eq 'localhost' || $CONFIG_DEFAULTS{'POSTGRES_SERVER'} eq '127.0.0.1' ) 
+{
+    $pg_local_connection = 1;
+} 
+elsif ($pg_only) 
+{
+    print "In the --pg_only mode the script can only be run LOCALLY,\n";
+    print "i.e., on the server where PostgresQL is running, with the\n";
+    print "Postgres server address as localhost - \"127.0.0.1\".\n";
+    exit 1;
+}
+
+### 3a. CHECK FOR USER postgres: (NO LONGER USED!)
+
+###print "\nChecking system user \"postgres\"... ";
+
+###my $POSTGRES_SYS_NAME = "postgres";
+###$POSTGRES_SYS_UID = ( getpwnam("postgres") )[2];
+
+# 3b. LOCATE THE EXECUTABLES, FOR jq AND psql:
+
+my $sys_path = $ENV{'PATH'};
+my @sys_path_dirs = split( ":", $sys_path );
+$psql_exec = "";
+
+
+# 4. CONFIGURE POSTGRES:
+
+# 4a. BUT FIRST, CHECK IF WE CAN TALK TO POSTGRES AS THE ADMIN:
+$psql_exec = "PGPASSWORD=" . $CONFIG_DEFAULTS{'POSTGRES_ADMIN_PASSWORD'} . "; export PGPASSWORD; " ;
+my $psql_admin_exec = "PGPASSWORD=" . $CONFIG_DEFAULTS{'POSTGRES_PASSWORD'} . "; export PGPASSWORD; ";
+
+if (system( $psql_admin_exec . "/usr/bin/psql -h " . $CONFIG_DEFAULTS{'POSTGRES_SERVER'} . " -U postgres -d postgres -c 'SELECT * FROM pg_roles' > /dev/null 2>&1" ) ) 
+{
+    # No, we can't. :(
+    if ($pg_local_connection) 
+    {
+    # If Postgres is running locally, this is a fatal condition. 
+    # We'll give them some (potentially) helpful pointers and exit.
+    print "Nope, I haven't been able to connect to the local instance of PostgresQL as the admin user.\n";
```

Comment:
Probably makes sense to delete the pointers that don't apply to the target system.